### PR TITLE
Exception when running inappropriate file as a script, fix 2839

### PR DESCRIPTION
**Pull request by kira0204**
@kajojify and I had a discussion on this and this was a very difficult choice for us to make.
With the solution that this PR brings forward, although it's a neat solution but it might disturb the user experience when the error from his/her script is not a `SyntaxError` because it won't be informative enough

Another solution is not a neat solution which is proposed in #2839 comments.

### CODE REVIEW

File: mitmproxy/addons/script.py [hunk 1]
```diff
@@ -30,9 +30,11 @@ def load_script(path: str) -> types.ModuleType:
         loader.exec_module(m)
         if not getattr(m, "name", None):
             m.name = path  # type: ignore
-        return m
+    except Exception as e:
+        ctx.log.error(str(e))
```
Comment (cortesi) [inline mitmproxy/addons/script.py]: A few comments here. First, the structure of this except bothers me - we will see an unbound local error for the m variable if there's an error somewhere before m is assigned. I think the right approach is to change the signature of the function to possibly return None, and then to make sure that m has a default value of None before entering the except handler. Second, we know the path here, so we should add that to the error message being logged, so the user knows which file had the error.



I think a better approach is to add another try/except just around the exec_module

File: test/mitmproxy/data/addonscripts/recorder/error.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+impotr recorder
```
Comment (cortesi) [inline test/mitmproxy/data/addonscripts/recorder/error.py]: Let's make it more obvious that this is intended to be an invalid file with a syntax error, just so I don't accidentally fix it someday. :)

Comment (kira0204) [reply test/mitmproxy/data/addonscripts/recorder/error.py]: Sure thankyou :) I will write this in the comments that this is an intended errors :)

File: mitmproxy/addons/script.py (hunk 1 above)
Comment (kira0204) [reply mitmproxy/addons/script.py]: Thanks for the extensive review :) When I am printing the error as `e` , it also prints the filename and the lineno with it like `invalid syntax(example.py, line 4)`, then this is only for SyntaxError and I couldn't find a way to print the filename for other errors like `ValueError` mentioned in the original error. Even the traceback module didn't help :(

**Comment by kira0204** [comment]
Closed because #2962

**Pull request closed without merging**